pkg/cmd/release/update/update.go

```diff
@@ -0,0 +1,186 @@
+	opts := &UpdateOptions{
+		IO:         f.IOStreams,
+		HttpClient: f.HttpClient,
+	}
+
+	var notesFile string
+
+	cmd := &cobra.Command{
+		DisableFlagsInUseLine: true,
+
+		Use:   "update <release_id>",
+		Short: "Edit a release",
+		Example: heredoc.Doc(`
+			Publish a release that was previously a draft
+			$ gh release update 63899317 --draft=false
+
+			Mark a release as a prerelease
+			$ gh release update 63899317 --prerelease
+
+			Mark a release as a full release
+			$ gh release update 63899317 --prerelease=false
+
+			Update the target commitish of a release
+			$ gh release update 63899317 --target 97ea5e77b4d61d5d80ed08f7512847dee3ec9af5
+
+			Update the name of the tag
+			$ gh release update 63899317 --tag v9.8.7
+
+			Update the name/title of a release
+			$ gh release update 63899317 --title 'Some new title'
+
+			Update the release notes
+			$ gh release update 63899317 --notes 'Some short release notes'
+
+			Update the release notes from the content of a file
+			$ gh release update 63899317 --notes-file /path/to/release_notes.md
+		`),
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.ReleaseId = args[0]
+
+			opts.BaseRepo = f.BaseRepo
+			opts.RepoOverride, _ = cmd.Flags().GetString("repo")
+
+			err := parseOptionalBoolFlags(cmd, opts)
+			if err != nil {
+				return err
+			}
+
+			opts.BodyProvided = cmd.Flags().Changed("notes")
+			if notesFile != "" {
+				b, err := cmdutil.ReadFile(notesFile, opts.IO.In)
+				if err != nil {
+					return err
+				}
+				opts.Body = string(b)
+				opts.BodyProvided = true
+			}
+
+			if runF != nil {
+				return runF(opts)
+			}
+			return updateRun(opts)
+		},
+	}
+
+	cmd.Flags().Bool("draft", false, "Save the release as a draft instead of publishing it")
+	cmd.Flags().Bool("publish", false, "Publish the release")
+	cmd.Flags().Bool("prerelease", false, "Mark the release as a prerelease")
+	cmd.Flags().StringVar(&opts.Target, "target", "", "Target `branch` or full commit SHA (default: main branch)")
+	cmd.Flags().StringVar(&opts.TagName, "tag", "", "The name of the tag")
+	cmd.Flags().StringVarP(&opts.Name, "title", "t", "", "Release title")
+	cmd.Flags().StringVarP(&opts.Body, "notes", "n", "", "Release notes")
+	cmd.Flags().StringVarP(&notesFile, "notes-file", "F", "", "Read release notes from `file` (use \"-\" to read from standard input)")
+
+	return cmd
+}
+
+func updateRun(opts *UpdateOptions) error {
+	httpClient, err := opts.HttpClient()
+	if err != nil {
+		return err
+	}
+
+	baseRepo, err := opts.BaseRepo()
+	if err != nil {
+		return err
+	}
+
+	params := getParams(opts)
+
+	if len(params) == 0 {
+		return errors.New("nothing to update")
+	}
+
+	updatedRelease, err := updateRelease(httpClient, baseRepo, opts.ReleaseId, params)
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintf(opts.IO.Out, "%s\n", updatedRelease.URL)
+
+	return nil
+}
+
+func parseOptionalBoolFlags(cmd *cobra.Command, opts *UpdateOptions) error {
```

Comment:
You can avoid doing any of this by defining the flag using `cmdutil.NilBoolFlag()` in the first place.

https://github.com/cli/cli/blob/6783509d3bfdf55e1007dbedf34f2bdcd9fd1ec1/pkg/cmd/repo/edit/edit.go#L138-L151